R/get_ref.R

```diff
@@ -51,8 +51,19 @@ get_ref_min_incl_cran.remote_ref_github <- function(remote_ref, op = "", op_ver
 #' Check if package is available on CRAN.
 #' @importFrom pkgcache meta_cache_list
 #' @keywords internal
-check_if_on_cran <- function(remote_ref) {
-  nrow(pkgcache::meta_cache_list(remote_ref$package)) > 0
+#'
+#' @examplesIf Sys.getenv("R_USER_CACHE_DIR", "") != ""
+#' verdepcheck:::check_if_on_cran(list(package = "magrittr"))
+#' verdepcheck:::check_if_on_cran(list(package = "magrittr"), list(op = ">=", op_ver = "0.5.0"))
+#' verdepcheck:::check_if_on_cran(list(package = "magrittr"), list(op = ">=", op_ver = "9999.9.99"))
+#' verdepcheck:::check_if_on_cran(list(package = "magrittr"), list(op = "<", op_ver = "0.0.0"))
+check_if_on_cran <- function(remote_ref, version = NULL) {
+  cran_listings <- pkgcache::meta_cache_list(remote_ref$package)
+  if (is.null(version) || NROW(cran_listings) == 0) {
+    return(NROW(cran_listings) > 0)
+  }
+  # Check if minimum version exists on CRAN
+  NROW(filter_valid_version(cran_listings$version, version$op, version$op_ver)) > 0
```

Comment:
Adds `version` parameter to avoid turning a remote GitHub reference to a CRAN if that particular version doesn't exist on CRAN.

This is useful for a scenario where the dependency has been bumped up, but that version has not been released yet on CRAN.

Reply:
Very good change. Thank you!